dbg/dbg.c ,crash while building on OpenBSD

`apple2 - dbg/dbg.c`
`gmake[2]: *** [Makefile:283: ../libwrk/apple2/dbg.o] Segmentation fault (core dumped)`
`gmake[1]: *** [Makefile:118: apple2] Error 2`
`gmake: *** [Makefile:7: all] Error 2`
 
`~/source/cc65/libsrc $ gdb ../bin/cc65 cc65.core`
`GNU gdb 6.3`
`Copyright 2004 Free Software Foundation, Inc.`
`GDB is free software, covered by the GNU General Public License, and you are`
`welcome to change it and/or distribute copies of it under certain conditions.`
`Type "show copying" to see the conditions.`
`There is absolutely no warranty for GDB.  Type "show warranty" for details.`
`This GDB was configured as "amd64-unknown-openbsd6.7"...`
`Core was generated by  cc65'.`
`Program terminated with signal 11, Segmentation fault.`
`Loaded symbols for /home/frakswe/source/cc65/bin/cc65`
`Reading symbols from /usr/lib/libm.so.10.1...done.`
`Loaded symbols for /usr/lib/libm.so.10.1`
`Reading symbols from /usr/lib/libc.so.96.0...done.`
`Loaded symbols for /usr/lib/libc.so.96.0`
`Reading symbols from /usr/libexec/ld.so...Error while reading shared library symbols:`
`Dwarf Error: wrong version in compilation unit header (is 4, should be 2) [in module /usr/libexec/ld.so]`
`#0  strlen () at /usr/src/lib/libc/arch/amd64/string/strlen.S:125`
`125     /usr/src/lib/libc/arch/amd64/string/strlen.S: No such file or directory.`
`      in /usr/src/lib/libc/arch/amd64/string/strlen.S`
`(gdb) q`

What is strlen.S ?; 
[https://github.com/openbsd/src/blob/master/lib/libc/arch/amd64/string/strlen.S](url)
 
`       _ALIGN_TEXT`
`1:`
`        movq    (%rax),%rdx             /* get bytes to check */`


Testing cc65 on another random file does not crash:
 
`~/source/cc65/libsrc/apple2 $ ../../bin/cc65 closedir.c -o closedir.o`
`~/source/cc65/libsrc/apple2 $ ls closedir*`
`closedir.c  closedir.o`


Building cc65 with clang or gcc v8 , same result.

Why this?

--- a/src/cc65/codeent.c
+++ b/src/cc65/codeent.c
@@ -585,8 +585,7 @@ const char* MakeHexArg (unsigned Num)
 void PreparseArg (CodeEntry* E)
 /* Parse the argument string and memorize the result for the code entry */
 {
-    StrBuf B;
-    SB_InitFromString (&B, xmalloc (strlen (E->Arg) + 1));
+    StrBuf B = AUTO_STRBUF_INITIALIZER;
 
     /* Parse the argument string */
     if (ParseOpcArgStr (E->Arg, &E->ArgInfo, &B, &E->ArgOff)) {

